Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol CacheDirectoriesProviding: AnyObject {
+    /// Returns the cache directory
+    var cacheDirectory: AbsolutePath { get }
+
+    /// Directory where the projects generated for automation are located.
+    var projectsCacheDirectory: AbsolutePath { get }
```

Comment:
afaict this is not used anywhere (probably a mistake by me). I think it's safe to delete this.